pkg/distributor/distributor.go

```diff
@@ -137,6 +151,8 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.ShardByAllLabels, "distributor.shard-by-all-labels", false, "Distribute samples based on all labels, as opposed to solely by user and metric name.")
 	f.StringVar(&cfg.ShardingStrategy, "distributor.sharding-strategy", util.ShardingStrategyDefault, fmt.Sprintf("The sharding strategy to use. Supported values are: %s.", strings.Join(supportedShardingStrategies, ", ")))
 	f.BoolVar(&cfg.ExtendWrites, "distributor.extend-writes", true, "Try writing to an additional ingester in the presence of an ingester not in the ACTIVE state. It is useful to disable this along with -ingester.unregister-on-shutdown=false in order to not spread samples to extra ingesters during rolling restarts with consistent naming.")
+	f.Float64Var(&cfg.MaxIngestionRate, "distributor.max-ingestion-rate", 0, "Max ingestion rate (samples/sec) that this distributor will accept. This limit is per-distributor, not per-tenant. Additional push requests will be rejected. Current ingestion rate is computed as exponentially weighted moving average, updated every second. 0 = unlimited.")
```

Comment:
In the ingester the config is `-ingester.instance-limits.max-ingestion-rate` (has `instance-limits`). What do you think if we keep it specular in distributors too?

_Same comment applies to `max-inflight-push-requests` too._

Reply:
Do you suggest to introduce `instance_limits` field in YAML as well?

Reply:
> Do you suggest to introduce instance_limits field in YAML as well?

I've done it that way.

Reply:
Yes, to keep consistency between CLI and YAML.